app/Models/Signup.php

```diff
@@ -65,21 +65,66 @@ public function posts()
     }
 
     /**
-     * Get the visible posts associated with this signup.
+     * Get the visible posts associated with this signup, optionally by type.
      *
+     * @param string Optional param to specify post type.
      * @return \Illuminate\Database\Eloquent\Relations\HasMany
      */
-    public function visiblePosts()
+    public function visiblePosts($type = null)
     {
         if (! is_staff_user()) {
-            return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+            if ($type === null) {
+                return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+                                                  ->orWhere('northstar_id', auth()->id())
+                                                  ->with('tags');
+            }
+
+            return $this->hasMany(Post::class)->whereIn('type', array_values($type))
+                                              ->where('status', '=', 'accepted')
                                               ->orWhere('northstar_id', auth()->id())
-                                              ->with('tags');
+                                              ->with('tags')
+                                              ->get();
+            // return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+            //                                   ->orWhere('northstar_id', auth()->id())
+            //                                   ->whereIn('type', array_values($type))
+            //                                   ->with('tags')
+            //                                   ->get();
+
+            // return $this->hasMany(Post::class)->where(function ($q) use ($type) {
+            //     $q->whereIn('type', array_values($type))
+            //       ->where('status', '=', 'accepted')
+            //       ->orWhere('northstar_id', auth()->id())
+            //       ->with('tags')
+            //       ->get();
+            // });
+
+            // return $this->hasMany(Post::class)->whereIn('type', array_values($type))
+            //                                   ->where(function($q) {
+            //                                     $q->where('status', '=', 'accepted')
+            //                                       ->orWhere('northstar_id', auth()->id());
+            //                                   })
+            //                                   ->with('tags')
+            //                                   ->get();
+
```

Comment:
@DFurnes here are some examples of how I tried to do the parameter grouping! Still didn't work though and returned voter-reg posts when hitting `api/v3/signups?include=posts:type(text|photo)` :(

Reply:
omg I am an idiot - when testing I must have made a mistake - this actually _does_ work!!